### PR TITLE
Update vuln deps, CRA -> vite, add unit testing support

### PR DESCRIPTION
**Pull request by chao-mu**
* Update several outdated or outright abandoned dependencies
* Migrate away from dead CRA to Vite
* Support unit testing via vitest
* Update App.test.tsx to work with Dojo code (Appears to have been generated at project creation)
* Add some helpers to package.json

`npm run test` runs all tets. 
`npm run test:e2e` runs cypress only.
`npm run test:unit` runs only unit tests.

Please hold off on merging for now. I want to give it one more pass over.

### CODE REVIEW

File: frontend/index.html [hunk 1]
```diff
@@ -2,25 +2,25 @@
...
+        <link rel="apple-touch-icon" href="/android-chome-192x192.png" />
         <!--
       manifest.json provides metadata used when your web app is installed on a
       user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
     -->
-        <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+        <link rel="manifest" href="/manifest.json" />
         <!--
-      Notice the use of %PUBLIC_URL% in the tags above.
+      Notice the use of  in the tags above.
```
Comment (jackstenglein) [inline frontend/index.html:19]: This comment doesn't make sense now that `%PUBLIC_URL%` is no longer in the url. Also, why was that removed? Is that only a CRA thing?

File: frontend/package.json [hunk 2]
```diff
@@ -2,71 +2,80 @@
     "name": "frontend",
     "version": "0.1.0",
     "private": true,
+    "type": "module",
     "dependencies": {
-        "@aldabil/react-scheduler": "^2.7.26",
-        "@aws-amplify/auth": "^5.6.4",
+        "@aldabil/react-scheduler": "^2.8.8",
```
Comment (jackstenglein) [inline frontend/package.json:7]: Please revert the upgrade to react-scheduler. He introduced a breaking change to the way timezone overriding works in 2.8 and I haven't had time to refactor the Dojo code to work with that.

File: frontend/package.json [hunk 3]
```diff
@@ -2,71 +2,80 @@
...
-        "@mui/x-date-pickers": "^6.19.3",
-        "@testing-library/jest-dom": "^6.4.2",
-        "@testing-library/react": "^14.2.1",
-        "@testing-library/user-event": "^14.4.3",
-        "@types/node": "^20.11.24",
-        "@types/react": "^18.2.61",
-        "@types/react-dom": "^18.2.19",
+        "@mui/x-data-grid-pro": "^6.19.10",
+        "@mui/x-date-pickers": "^6.19.9",
+        "@svgr/core": "^8.1.0",
```
Comment (jackstenglein) [inline frontend/package.json:25]: What is this @svgr/core package (and the svgr/plugin-jsx below it) used for?

File: frontend/src/App.test.tsx [hunk 4]
```diff
@@ -1,9 +1,9 @@
-import React from 'react';
 import { render, screen } from '@testing-library/react';
+import { expect, test } from 'vitest';
 import App from './App';
 
-test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+test('renders learn react link', async () => {
```
Comment (jackstenglein) [inline frontend/src/App.test.tsx:5]: Can we change the name of this test as well?

File: frontend/package.json [hunk 5]
```diff
@@ -2,71 +2,80 @@
...
+        "@svgr/plugin-jsx": "^8.1.0",
+        "@types/node": "^20.12.7",
+        "@types/react": "^18.2.75",
+        "@types/react-dom": "^18.2.24",
+        "@vitejs/plugin-react": "^4.2.1",
         "@widgetbot/react-embed": "^1.9.0",
-        "aws-amplify": "^5.3.10",
+        "aws-amplify": "^5.3.18",
+        "axios": "^1.6.8",
+        "chart.js": "^4.4.2",
```
Comment (jackstenglein) [inline frontend/package.json:34]: We can remove chart.js right? You were going to replace react-charts with this but then didn't need to?

**Review by jackstenglein — CHANGES_REQUESTED** [review]

File: frontend/index.html (hunk 1 above)
Comment (chao-mu) [reply frontend/index.html:19]: Will look into this and get back. This file was moved/modified by a conversion tool and I only did a smell test/actual testing.

Comment (chao-mu) [reply frontend/index.html:19]: Okay, tested/compared the builds from dev, production, and this PR. This is no longer needed and is populated automatically for us by Vite.

From the migration guide I followed (I refreshed my memory and these steps I did manually):

Vite [automatically resolves](https://vitejs.dev/guide/#index-html-and-project-root) URLs inside index.html, so there's no need for %PUBLIC_URL% placeholders. You can do a search and replace inside your index.html file for this. Be sure to remove all occurrences.
